js/gdax.js

```diff
@@ -269,20 +276,39 @@ module.exports = class gdax extends Exchange {
                 'currency': market['quote'],
             };
         }
+        let type = undefined;
+        if ('liquidity' in trade) {
+            type = (trade['liquidity'] == 'T') ? 'Taker' : 'Maker';
+        }
         return {
-            'id': trade['trade_id'].toString (),
+            'id': trade['trade_id'].toString(),
+            'order': trade['order_id'].toString(),
             'info': trade,
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
             'symbol': symbol,
-            'type': undefined,
+            'type': type,
             'side': side,
-            'price': parseFloat (trade['price']),
-            'amount': parseFloat (trade['size']),
+            'price': parseFloat(trade['price']),
+            'amount': parseFloat(trade['size']),
             'fee': fee,
         };
     }
 
+    async fetchMyTrades(symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets();
+        let market = undefined;
+        let request = {}
+        if (symbol) {
+            market = this.market(symbol);
+            request['product_id'] = this.marketId(symbol)
+        }
+        if (limit)
+            request['limit'] = limit;
+        let response = await this.privateGetFills(this.extend(request, params));
```

Comment:
I would add [this](https://github.com/ccxt/ccxt/pull/1334/commits/fc3cf3ce42bd989fedd7a68cd2a574efd26d4497#diff-9453f5a8dee427abd6a1b4a0aed6a819R460),

```js
if (since)
    throw new NotSupported (this.id + ' can currently only return the most recent 100 trades'
        + 'due to their admittedly "unintuitive" pagination API - https://docs.gdax.com/#pagination');
```

Reply:
That multiline expression breaks Python syntax, therefore all expressions should be basic one-liners, mostly.